internal/pkg/descriptor/descriptor.go

```diff
@@ -0,0 +1,171 @@
+// Copyright 2019 The Operator-SDK Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package descriptor
+
+import (
+	"os"
+	"path"
+	"path/filepath"
+	"strings"
+
+	"github.com/operator-framework/operator-sdk/internal/util/projutil"
+
+	olmapiv1alpha1 "github.com/operator-framework/operator-lifecycle-manager/pkg/api/apis/operators/v1alpha1"
+	"github.com/pkg/errors"
+	log "github.com/sirupsen/logrus"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/gengo/parser"
+	"k8s.io/gengo/types"
+)
+
+// GetCRDDescriptorForGVK parses type and struct field declaration comments on
+// API types to populate a csv's spec.customresourcedefinitions.owned fields
+// for a given API identified by Group, Version, and Kind in apisDir.
+// TODO(estroz): support ActionDescriptors parsing/setting.
+func GetCRDDescriptorForGVK(apisDir string, crdDesc *olmapiv1alpha1.CRDDescription, gvk schema.GroupVersionKind) error {
+	specType, statusType, pkgTypes, found, err := findTypesForGVK(apisDir, gvk)
+	if err != nil {
+		return err
+	}
+	if !found {
```

Comment:
I was going to suggest that we move the check for `found` and skipping CRD descriptor parsing out from `findTypesForGVK()` since that seems like the first thing we should do and it would reduce the return values for `findTypesForGVK()`. Also make it more obvious what found is.

But when i wrote it out, I'm not so sure it looks all that simple:
```Go
func GetCRDDescriptorForGVK(apisDir string, crdDesc *olmapiv1alpha1.CRDDescription, gvk schema.GroupVersionKind) error {
	group = strings.Split(group, ".")[0]
	apiDir := filepath.Join(apisDir, group, gvk.Version)
	if _, err := os.Stat(apiDir); err != nil {
		if os.IsNotExist(err) {
			log.Infof("API directory %s does not exist. Skipping CSV annotation parsing for API %s.", apiDir, gvk)
			return nil
		}
		return err
	}

	specType, statusType, pkgTypes, err := findTypesForGVK(apisDir, gvk)
	if err != nil {
		return err
	}
	...
```

So I guess it's splitting hairs on whether the above is simpler to follow. WDYT?

At the very least if we keep `findTypesForGVK()` as it is, can you add godocs for it since there's quite a number of return values for it.

Reply:
Its just a matter of where clutter is organized. This is what your suggestion would look like (if I'm thinking about it right):
```go
func GetCRDDescriptorForGVK(apisDir string, crdDesc *olmapiv1alpha1.CRDDescription, gvk schema.GroupVersionKind) error {
	group := gvk.Group
	if strings.Contains(group, ".") {
		group = strings.Split(group, ".")[0]
	}
	apiDir := filepath.Join(apisDir, group, gvk.Version)
	universe, err := getTypesFromDir(apiDir)
	if err != nil {
		if os.IsNotExist(err) {
			log.Infof("API directory %s does not exist. Skipping CSV annotation parsing for API %s.", apiDir, gvk)
			return nil
		}
		return err
	}
	specType, statusType, pkgTypes, err := findTypesForGVK(apiDir, universe, gvk)
	if err != nil {
		return err
	}
	...
```

Reply:
There should be godocs for all of these anyway.

Reply:
Yes you're right, it's just moving it around. My initial thought process was to move it up front so it's clear that we skip early before calling a helper.
But given the clutter, it's fine either way.

`findTypesForGVK()` and `getTypesFromDir()` don't have godocs atm so if you can add that then it should be fine to leave this as is.